### PR TITLE
Profile Page Anchors

### PR DESCRIPTION
**Pull request by CMS-Falcon**
Added the profile page anchors. I'm a bit new to git, so hopefully I'm doing this correctly :)

### CODE REVIEW

File: resources/assets/coffee/react/profile-page/extra.coffee [hunk 1]
```diff
@@ -37,6 +37,29 @@ class ProfilePage.Extra extends React.Component
     osu.pageChange()
     @_modeScan()
 
+    if window.location.hash
+      url = window.location.href
+      anchor = url.substring(url.indexOf('#') + 1)
+      mode = ''
+      switch anchor
+        when 'me!'
```
Comment (peppy) [inline resources/assets/coffee/react/profile-page/extra.coffee]: this should definitely not be hardcoded here.

**Comment by CMS-Falcon** [comment]
Just to clarify, do you mean that piece of code belongs somewhere else, or that there is a problem with the way it is written (or both)?

**Comment by peppy** [comment]
As in you shouldn't be adding redundant copies of header names/identifiers that could change at any point. Get it from its source (in the php classes).

**Comment by CMS-Falcon** [comment]
Alright, will do. Thanks

**Comment by CMS-Falcon** [comment]
I think this is what you were referring to, let me know if I'm still off.

**Comment by peppy** [comment]
Still missing the point.. We have a list of all sections already, so you shouldn't be duplicating it redundantly:

**Comment by CMS-Falcon** [comment]
Yeah, I saw that list and for some reason thought that wasn't what I needed to do. I'll just blame the lack of sleep. Thanks for bearing with me, if there's anything else missing I'll try to get to it tomorrow.

**Comment by peppy** [comment]
See PR for a more correct implementation. Closing this one!

**Pull request closed without merging**